mmv1/third_party/terraform/resources/resource_compute_router_interface.go.erb

```diff
@@ -137,6 +145,22 @@ func resourceComputeRouterInterfaceCreate(d *schema.ResourceData, meta interface
 
 	iface := &compute.RouterInterface{Name: ifaceName}
 
+	if riVal, ok := d.GetOk("redundant_interface"); ok {
+		ifaceFound := false	
```

Comment:
I'm not sure I understand the purpose of this section, from L148 -> L159. Could it be redundant with the search for a duplicate interface in L139 -> L144?

Reply:
Hello @rileykarson !

I don't think it's redundant, but maybe I'm missing something.
What these lines do is to check that the name of the redundant interface given as argument is an existing interface of the router.

Reply:
Hello @rileykarson. Could you please help me to close this? I need to send another patch which will need to rebase on this one. Thanks!

Reply:
Oh, I understand now, thanks! The variable naming mixed me up, this checks the _redundant_ interface exists.

I think it'd be preferable to omit the check here, actually, and pass it off to the API. Clientside validations in Terraform are valuable when they're done at plan time- effectively, when they're *not* part of the CRUD methods- or when they save significant amounts of time (at least tens of seconds). The API's (hopefully) gonna have a more complete, and accurate error message, and should return pretty quickly.